### PR TITLE
Closes #1838 First Dummy Issue

### LINKED ISSUE #1838: General Issue: Dependencies conflict with tidyverse recomendations
### Background Information

Currently dependencies on external functions are handled via importing the functions into the namespace in the `admiral-package.R` script.

Hadley's book on packages recommends not to do this, and instead to write `package::function(`. Some example reasons why `::` is preferred are;

- It makes it clear which functions are from this package, and which are external. This makes it easier to onboard to the codebase, debug the codebase, and generally understand what a function is doing
- There is no risk of namespace conflicts
- You don't end up with mismatches as the package evolves and dependencies are dropped, as dependencies are called where they are used, rather than in a separate file

He does though note 3 exceptions where it can be useful to import,

1. you use the function a lot (and it's clear it comes from somewhere else - e.g. for admiral `mutate` or `left_join` will obviously be `dplyr` to devs and users).
2. it's a loop, as it's slower - but only noticeably if running millions of times
3. it's an operator, e.g. `%--%` from lubridate

As a side note, it's more redundant rather than a bug - but there is least one case where we accidentally have done both. e.g. [in this line of code](https://github.com/pharmaverse/admiral/blame/6dead5ad9b9fe529d8b459ae718f8f35cb52296f/R/derive_date_vars.R#L369) the user wrote `stringr::str_match` despite the fact it's imported into the namespace

ref: https://r-pkgs.org/dependencies-in-practice.html#in-code-below-r

### Definition of Done

Consider a switch to following Hadley's best practices and move to `::`

### PR DESCRIPTION
**Pull request by ngiangre**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by ngiangre** [comment]
I couldn't add the issue under the development section for some reason - it was grayed out.

File: NEWS.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## New Features
 
+- added my_first_function
```
Comment (bms63) [inline NEWS.md:5]:
```suggestion
- added my_first_function (#1838)
```

Make sure to add issue number in news blurb

**Comment by bms63** [comment]
> I couldn't add the issue under the development section for some reason - it was grayed out.

Maybe it has something to do with others using this issue in their PRs as well??

All good!  Any questions or concerns?  If not, please close this PR.

**Comment by ngiangre** [comment]
no questions - very smooth process thanks to the great docs. Thanks!
